### ISSUE TITLE
Deeply nested keys are not passed correctly to afterSave triggers

### ISSUE THREAD

**Issue by mstniy** [parse-community/parse-server#7384]
### New Issue Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).
- [x] I can reproduce the issue with the [latest version of Parse Server](https://github.com/parse-community/parse-server/releases). 

### Issue Description
afterSave triggers do not receive the correct updated object, if a deeply nested key was updated.

### Steps to reproduce

1. Create an afterSave trigger for a class
2. Create a document with a deeply nested key:
```
const obj = new Parse.Object('GameScore');
obj.set('a', {'b':{'c':0}});
await obj.save();
```
3. Change the deeply nested key
```
obj.set('a.b.c', 1);
await obj.save();
```

### Actual Outcome
The afterSave triggers receives a wrong object (in request.object)

### Expected Outcome
The field `a` of the received object should be `{'b':{'c':0}}`

### Failing Test Case / Pull Request

- [ ] 🤩 I submitted a PR with a fix and a test case.
- [ ] 🧐 I submitted a PR with a failing test case.

###  Environment

Server
- Parse Server version: Latest commit as of 12 May 2021, `51e0800`
- Operating system: `any`
- Local or remote host (AWS, Azure, Google Cloud, Heroku, Digital Ocean, etc): `local`

Database
- System (MongoDB or Postgres): `MongoDB`
- Database version: `v4.4.5`
- Local or remote host (MongoDB Atlas, mLab, AWS, Azure, Google Cloud, etc): `local`

Client
- SDK (iOS, Android, JavaScript, PHP, Unity, etc): `any`
- SDK version: `any`

**Comment by mtrezza** [comment]
Thanks for reporting.

Can you specify what exactly that wrong object is according to your example?

> The afterSave triggers receives a wrong object (in request.object)

**Comment by mstniy** [comment]
@mtrezza The trigger gets `{"a":{"b":1}`
Inspecting the code, it completely ignores any depth further than the first two, so in this example, `a` and `b` make it, but `c` gets dropped.

**Comment by mtrezza** [comment]
Thanks for the details.

**Comment by mstniy** [comment]
The issue remains even after #7005
[https://github.com/Ocell-io/parse-server/tree/deep_7384_test_only](https://github.com/Ocell-io/parse-server/tree/deep_7384_test_only) demonstrates this.
#7385 is the approach our team is still using, as we make heavy use of afterSave triggers and deeply nested keys, but it doesn't seem to work with Postgre, and we don't have the resources to investigate into it. Thus, I have marked it as a draft.

**Comment by ArkeshGKalathiya** [comment]
Possibly because of this bug, live queries also don't work properly, and there is no work around for that.
